### DISCUSSION TITLE
Online conversion

### DISCUSSION THREAD

**Discussion by guillaumerose** [Audiveris/audiveris#551] [Show and tell]
Hello,

My son is learning French horn and Audiveris helped me a lot when I had to import a few scores in MuseScore. I couldn't find a way to convert a score without having to install Audiveris so I created this web service.

https://scorescanner.guillaumerose.fr/

Few improvements are still needed: add a proper queue, maybe the ability to rate the output of the conversion, show the license, etc.

It's a start. Let me know what you think!

**Comment by Altonss** [comment]
Wooow this looks really great!! Thanks for that web service!
But I noticed Audiveris is licensed under AGPL-3.0 license, so to comply with that license it would be great (necessary) to release the source code of the web service also under that same license :)

**Reply by guillaumerose** [reply]
Yes, you're right. Let's fix that right now. The code is here: https://github.com/guillaumerose/audiveris-service

**Reply by Altonss** [reply]
That's great, thank you!
It's really a good idea to have it available as a web service for people who might not be so tech-savy to install and play around with the whole audiveris.

**Comment by hbitteur** [comment]
@guillaumerose Congratulations for this service which brings OMR to everybody.

My understanding is that you rely on Audiveris OMR engine to perform score transcription. It's OK for rather simple and good-quality inputs, but may be rather disappointing for poor-quality / crowded inputs. For those ones, manual corrections will be mandatory, requiring the use of Audiveris GUI beside the OMR engine, and this would somehow defeat the simplicity and ease of use of your service.

But anyway, thanks for your initiative which is already a major step forward.

**Reply by guillaumerose** [reply]
Yes, you're right. I will add some explanations on the home page. 

Is there an indicator of quality internal to Audiveris that I can use? A report of the OMR?

I think this kind of service can help trying new algorithms (like the one in 6.x prototype - the user doesn't have to install the multi-GB model). It is also a good opportunity to build a test dataset.

**Reply by Altonss** [reply]
Well for building a test dataset, you would be required ask user consent to store and process the uploaded files for this purpose (GDPR :smiley: ). For now it's just a small project so it's fine, but if the service is going to grow, this consent would very much be needed.

**Comment by guillaumerose** [comment]
One month after this post, 5 scores were submitted (by 3 different users). This is quite unexpected! I will improve the tool then :)
